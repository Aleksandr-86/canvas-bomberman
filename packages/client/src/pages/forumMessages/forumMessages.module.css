.forum {
  width: 100vw;
  height: 100vh;
  background-image: url('../../assets/images/forumBackground.png');
  background-size: cover;
  background-repeat: no-repeat;
}

.forumWrapper {
  display: flex;
  flex-direction: column;
  gap: 30px;
  justify-content: flex-start;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background: hsl(120deg 10% 78% / 75%);
  backdrop-filter: blur(0.5vh);
}

.buttonWrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 20px;
  width: 600px;
}
