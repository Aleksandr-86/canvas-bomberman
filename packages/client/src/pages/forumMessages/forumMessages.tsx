import { Link } from 'react-router-dom'
import classes from './forumMessages.module.css'
import { Table } from '../../components/table/table'
const FORUM_MESSAGES = {
  headers: ['Автор', 'Тема', 'Дата'],
  rows: [
    {
      id: 1,
      postid: 1,
      author: 'Kan88',
      body: 'Как пройти первый уровень',
      date: '8.01.22',
    },
    {
      id: 2,
      postid: 1,
      author: 'sasha85',
      body: 'Я прошел на изи, надо лишь быстро двигать пальцами',
      date: '8.01.22',
    },
    {
      id: 3,
      postid: 2,
      author: 'Kan88',
      body: 'Подскажите пожалуйста как пройти второй уровень',
      date: '8.01.22',
    },
    {
      id: 4,
      postid: 2,
      author: 'sasha85',
      body: 'Я еще не прошел',
      date: '8.01.22',
    },
  ],
}
export const ForumMessages = () => {
  return (
    <div className={classes.forum}>
      <Table {...FORUM_MESSAGES} />
      <div className={classes.forumWrapper}>
        <Link to="/forum-new-message" className={classes.forumLink}>
          Создать сообщение
        </Link>
        <Link to="/" className={classes.forumLink}>
          назад
        </Link>
      </div>
    </div>
  )
}
