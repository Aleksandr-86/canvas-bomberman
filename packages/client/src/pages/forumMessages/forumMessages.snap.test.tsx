import * as renderer from 'react-test-renderer'
import { ForumMessages } from './forumMessages'
import { MemoryRouter } from 'react-router-dom'

describe(`Snapshot test page ForumMessages`, () => {
  test(`Should correct render (without props)`, () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[`/forum-messages`]}>
          <ForumMessages />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
