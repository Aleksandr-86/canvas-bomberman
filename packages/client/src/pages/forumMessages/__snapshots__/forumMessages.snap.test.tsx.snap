// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test page ForumMessages Should correct render (without props) 1`] = `
<div
  className="forum"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          className="tableTitle"
        >
          Автор
        </th>
        <th
          className="tableTitle"
        >
          Тема
        </th>
        <th
          className="tableTitle"
        >
          Дата
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="tableRow"
        >
          Kan88
        </td>
        <td
          className="tableRow"
        >
          Как пройти первый уровень
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
      <tr>
        <td
          className="tableRow"
        >
          sasha85
        </td>
        <td
          className="tableRow"
        >
          Я прошел на изи, надо лишь быстро двигать пальцами
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
      <tr>
        <td
          className="tableRow"
        >
          Kan88
        </td>
        <td
          className="tableRow"
        >
          Подскажите пожалуйста как пройти второй уровень
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
      <tr>
        <td
          className="tableRow"
        >
          sasha85
        </td>
        <td
          className="tableRow"
        >
          Я еще не прошел
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="forumWrapper"
  >
    <a
      className="forumLink"
      href="/forum-new-message"
      onClick={[Function]}
    >
      Создать сообщение
    </a>
    <a
      className="forumLink"
      href="/"
      onClick={[Function]}
    >
      назад
    </a>
  </div>
</div>
`;
